native/dgebrd.go

```diff
@@ -0,0 +1,92 @@
+package native
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dgebrd reduces a general m×n matrix A to upper or lower bidiagonal form B by
+// an orthogonal transformation: Q^T * A * P = B.
+func (impl Implementation) Dgebrd(m, n int, a []float64, lda int, d, e, tauQ, tauP, work []float64, lwork int) {
```

Comment:
(now in PR)

Given the complexity of the layout of elements in this, it's probably worth reflecting the documentation of DGEBRD.f about that. Maybe not in the doc comment, but it should be somewhere.